pkg/controller/daemon/daemoncontroller.go

```diff
@@ -763,6 +784,12 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 	if err != nil {
 		return fmt.Errorf("unable to retrieve ds %v from store: %v", key, err)
 	}
+	// DeepCopy so we don't mutate objects in the cache.
+	cp, err := api.Scheme.DeepCopy(ds)
```

Comment:
Are you referring to this DeepCopy? We can use a boolean to record if the controller's DeletionTimestamp is nil, then we don't need to modify the controller object.